guide_drafts/trait_derives.md

```diff
@@ -0,0 +1,927 @@
+Remember, `Queryable` structs represent the exact columns, value, 
+and ordering of your query's returned result,
+which we are violating here.
+`Queryable` is expecting a tuple that looks like `(i32, String, String, String)`,
+but we currently only have a tuple consisting of `(i32,)`.
+We need to add those `String` columns back for our code to compile again.
+See Rust's [data type docs] if you're unfamiliar with tuples or their syntax.
+
+[data type docs]: https://doc.rust-lang.org/1.21.0/book/second-edition/ch03-02-data-types.html#grouping-values-into-tuples
+
+## QueryableByName
+`Queryable is the trait you normally use with Diesel's query builder.
+If you're using the `sql_query` function,
+you will instead need to implement the `QueryableByName` trait.
+
+Diesel provides some escape hatches to execute raw SQL queries.
+The problem with this is that Diesel can't ensure type safety,
+which also means that you can't deserialize the raw query result into tuples or a struct.
+Adding [`#[derive(QueryableByName)]`](queryablebyname_doc) to your struct means 
+that struct will be the result of a raw SQL query using the [`sql_query`] function.
+
+[`sql_query`]: https://docs.diesel.rs/diesel/fn.sql_query.html
+[queryablebyname_doc]: https://docs.diesel.rs/diesel/query_source/trait.QueryableByName.html
+
+The implementation of `QueryableByName` assumes that each field on your struct
+has a certain SQL type.
+It makes these assumptions based on the annotations you add to your struct.
+If your `QueryableByName` struct references a single table,
+you may annontate that struct with `#[table_name="my_table"]`.
+`QueryableByName` will bind the struct fields to the SQL types it finds in your table's schema.
+
+You may also individually annotate each field on your struct
+with `#[sql_type="ColumnTypeHere"]`.
+If you're not using the `table_name` annotation,
+every field in your struct needs to have this annotation.
+When combining both `table_name` and `sql_type` annotations,
+Diesel will override any fields using `sql_type` and pick the rest from the table.
+
+`QueryableByName` also supports nested structs.
+If you have any fields whose type is a struct that also implements `QueryableByName`,
+you may add the `#[diesel(embed)]` annotation.
+
+### Example 
+
+```rust
+// File: src/models.rs
+
+use diesel::types::{Integer, Text};
+
+// We're defining the posts and users tables here
+// to illustrate the schema in the following example
+table! {
+    posts (id) {
+        id -> Int4,
+        user_id -> Int4,
+        title -> Varchar,
+        body -> Text,
+    }
+}
+
+table! {
+    users (id) {
+        id -> Int4,
+        first_name -> Varchar,
+        last_name -> Varchar,
+        email -> Varchar,
+    }
+}
+
+#[derive(Debug, Queryable)]
+pub struct User {
+    pub id: i32,
+    pub first_name: String,
+    pub last_name: String,
+    pub email: String,
+}
+
+// When not using the `table_name` annotation,
```

Comment:
…just above this line, so and maybe add a comment that it's required for the types to be in scope?